src/engine/condition.ts

```diff
@@ -1,45 +1,51 @@
 import * as reducers from './reducers';
 import * as matchers from './matchers';
 import * as filters from './filters';
-import { PageView, EngineCondition } from '../../types';
-import { isNumberArray, isStringArray } from '../utils';
+import {
+  PageView,
+  PageFeatureResult,
+  EngineCondition,
+  EngineConditionQuery,
+  ArrayIntersectsFilter,
+  VectorDistanceFilter,
+  CosineSimilarityFilter,
+  AudienceDefinitionFilter,
+  VectorQueryValue,
+} from '../../types';
+import {
+  isNumberArray,
+  isStringArray,
+  isArrayIntersectsFilter,
+  isVectorDistanceFilter,
+  isCosineSimilarityFilter,
+} from '../utils';
 
-const createCondition = (condition: EngineCondition) => (
-  pageViews: PageView[]
-): boolean => {
+const createCondition = (condition: EngineCondition<AudienceDefinitionFilter>) =>
+  (pageViews: PageView[]): boolean => {
   const { filter, rules } = condition;
   const filteredPageViews = filter.queries
-    .map((query) => {
-      return pageViews.filter((pageView) => {
-        const queryFeatures = pageView.features[query.property];
-
-        if (query.filterComparisonType === 'arrayIntersects') {
-          return (
-            !!queryFeatures &&
-            queryFeatures.version === query.version &&
-            isStringArray(queryFeatures.value) &&
-            filters.arrayIntersects(queryFeatures.value, query.value)
-          );
-        } else if (query.filterComparisonType === 'vectorDistance') {
-          return (
-            !!queryFeatures &&
-            queryFeatures.version === query.version &&
-            isNumberArray(queryFeatures.value) &&
-            filters.vectorDistance(queryFeatures.value, query.value)
-          );
-        } else if (query.filterComparisonType === 'cosineSimilarity') {
-          return (
-            !!queryFeatures &&
-            queryFeatures.version === query.version &&
-            isNumberArray(queryFeatures.value) &&
-            filters.cosineSimilarity(queryFeatures.value, query.value)
-          );
-        } else {
-          return true;
-        }
-      });
-    })
-    .flat();
+  .map((query) => {
+    return pageViews.filter((pageView) => {
+      const queryFeatures: PageFeatureResult = pageView.features[query.queryProperty];
+      return isArrayIntersectsFilter(query)
+        ? isArrayIntersects(queryFeatures, query)
+        : isVectorDistanceFilter(query)
+        ? isVectorDistanceLesserThanThreshold(queryFeatures, query)
+        : isCosineSimilarityFilter(query)
+        ? isCosineSimilarityLesserThanThreshold(queryFeatures, query)
+        : true;  /* TODO is this right?
```

Comment:
No this is wrong - I cannot think of why this would be here.

If none of the filters work we should not match by default.

Good spot 👍